kratos/solving_strategies/schemes/sensitivity_builder_scheme.h

```diff
@@ -569,16 +569,18 @@ class SensitivityBuilderScheme
 
         this->CalculateLocalSensitivity(rEntity, rResponseFunction,
                                         rSensitivityVector, rVariable, rProcessInfo);
-
-        auto& r_geometry = rEntity.GetGeometry();
-        if (rGPSensitivityVector.size() != r_geometry.PointsNumber()) {
-            rGPSensitivityVector.resize(r_geometry.PointsNumber());
-        }
-
-        for (unsigned int i = 0; i < r_geometry.PointsNumber(); ++i) {
-            rGPSensitivityVector(i) = mGlobalPointerNodalMap[r_geometry[i].Id()];
+        
+        if (rSensitivityVector.size() > 0) {
+            auto& r_geometry = rEntity.GetGeometry();
+            if (rGPSensitivityVector.size() != r_geometry.PointsNumber()) {
+                rGPSensitivityVector.resize(r_geometry.PointsNumber());
+            }
```

Comment:
`std::vector::resize` [performs this check internally](https://en.cppreference.com/w/cpp/container/vector/resize), so there's no need to duplicate it.

```suggestion
            rGPSensitivityVector.resize(r_geometry.PointsNumber());
```

Reply:
Thanks @matekelemen